static/js/robot.js

```diff
@@ -68,7 +98,191 @@ class RobotClass {
       // this.tapeLines.push(new TapeLine(27, 27, 115, 27))
     }
 
+    intersectRobotRef(obj, corners) {
+      // coordinates of the k_i vectors
+      let k1x = obj.botL[0] - corners.botL[0]; //x of the vector from botL of robot to botL of obstacle
+      let k1y = obj.botL[1] - corners.botL[1]; //figure it out from here...
+      let k2x = obj.topL[0] - corners.botL[0];
+      let k2y = obj.topL[1] - corners.botL[1];
+      let k3x = obj.topR[0] - corners.botL[0];
+      let k3y = obj.topR[1] - corners.botL[1];
+      let k4x = obj.botR[0] - corners.botL[0];
+      let k4y = obj.botR[1] - corners.botL[1];
+
+      // vector from botL to botR of robot
+      let ref1x = corners.botR[0] - corners.botL[0];
+      let ref1y = corners.botR[1] - corners.botL[1];
+
+      // vector from botL to topL of robot
+      let ref2x = corners.topL[0] - corners.botL[0];
+      let ref2y = corners.topL[1] - corners.botL[1];
+
+      // make the ref1 vector into a unit vector
+      let ref1mag = Math.sqrt(ref1x * ref1x + ref1y * ref1y);
+      ref1x = ref1x / ref1mag;
+      ref1y = ref1y / ref1mag;
+
+      // make the ref2 vector into a unit vector
+      let ref2mag = Math.sqrt(ref2x * ref2x + ref2y * ref2y);
+      ref2x = ref2x / ref2mag;
+      ref2y = ref2y / ref2mag;
+
+      let k1ref1ProjLen = k1x * ref1x + k1y * ref1y;
+      let k2ref1ProjLen = k2x * ref1x + k2y * ref1y;
+      let k3ref1ProjLen = k3x * ref1x + k3y * ref1y;
+      let k4ref1ProjLen = k4x * ref1x + k4y * ref1y;
+
+      let ref1inter = true;
+
+      if (this.findMax(k1ref1ProjLen, k2ref1ProjLen, k3ref1ProjLen, k4ref1ProjLen) >= ref1mag) {
+        if (this.findMin(k1ref1ProjLen, k2ref1ProjLen, k3ref1ProjLen, k4ref1ProjLen) >= ref1mag) {
+          ref1inter = false;
+        }
+      } else if (this.findMax(k1ref1ProjLen, k2ref1ProjLen, k3ref1ProjLen, k4ref1ProjLen) <= 0) {
+        if (this.findMin(k1ref1ProjLen, k2ref1ProjLen, k3ref1ProjLen, k4ref1ProjLen) <= 0) {
+          ref1inter = false;
+        }
+      }
+
+      let k1ref2ProjLen = k1x * ref2x + k1y * ref2y;
+      let k2ref2ProjLen = k2x * ref2x + k2y * ref2y;
+      let k3ref2ProjLen = k3x * ref2x + k3y * ref2y;
+      let k4ref2ProjLen = k4x * ref2x + k4y * ref2y;
+
+      let ref2inter = true;
+
+      if (this.findMax(k1ref2ProjLen, k2ref2ProjLen, k3ref2ProjLen, k4ref2ProjLen) >= ref2mag) {
+        if (this.findMin(k1ref2ProjLen, k2ref2ProjLen, k3ref2ProjLen, k4ref2ProjLen) >= ref2mag) {
+          ref2inter = false;
+        }
+      } else if (this.findMax(k1ref2ProjLen, k2ref2ProjLen, k3ref2ProjLen, k4ref2ProjLen) <= 0) {
+        if (this.findMin(k1ref2ProjLen, k2ref2ProjLen, k3ref2ProjLen, k4ref2ProjLen) <= 0) {
+          ref2inter = false;
+        }
+      }
+
+      return ref1inter && ref2inter;
+    }
+
+    intersectObjRef(obj, corners) {
```

Comment:
Could you add docstrings to this function and `intersectRobotRef()` describing what each function does? You can take a look at other `RobotClass` methods for a style reference.